libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -571,7 +579,22 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			else
 				args.emplace_back(convert(*arguments[i], *parameterTypes[i]).commaSeparatedList());
 
-		if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+		if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
+		{
+			solUnimplementedAssert(!functionType->bound(), "Internal calls to bound library functions are not yet implemented");
+
+			auto functionDef = dynamic_cast<FunctionDefinition const*>(memberAccess->annotation().referencedDeclaration);
+			solAssert(functionDef, "Libraries cannot have state variables");
+			solAssert(!functionDef->virtualSemantics(), "Libraries can neither inherit nor be inherited");
+
+			define(_functionCall) <<
```

Comment:
Can you join this with the next case? Basically have `if` and `else` that assign a FunctionDefinition pointer. If it is assigned, then generate the function call, else use the internal dispatch.

Reply:
Done.